### PR TITLE
feat: add docs for grid and layout proposal

### PR DESCRIPTION
**Pull request by sashuk**
[FX-3859](https://toptal-core.atlassian.net/browse/FX-3859)

### Description

This pull requests updates breakpoints and adds decision on layouts in Picasso and their compatibility with BASE design. Please see https://github.com/toptal/picasso/blob/FX-3859-add-grids-layout/docs/decisions/14-page-layout.md for the proposal.

### Development checks

- [N/A] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [N/A] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [N/A] Covered with tests
- [x] ⚠️ **Decision** is made and added to the file **before merge**

**Breaking change**

- [N/A] codemod is created and showcased in the changeset
- [N/A] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3859]: https://toptal-core.atlassian.net/browse/FX-3859?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/decisions/13-breakpoints.md [hunk 1]
```diff
@@ -33,26 +33,24 @@ sm:  480px  (small)
...
+Range         |   xs   |   sm   |   md   |   lg   |   xl
 ```
 
 ## Decision
 
 In order to adopt new breakpoints in Picasso, the following steps need to be taken:
 
 - update breakpoints in Picasso
-- extend Grid, RadioGroup, CheckboxGroup APIs to take into account new breakpoints and breakpoint ranges
+- [Grid item](https://picasso.toptal.net/?path=/story/layout-grid--grid), [Radio group](https://picasso.toptal.net/?path=/story/forms-radio--radio), and [Checkbox group](https://picasso.toptal.net/?path=/story/forms-checkbox--checkbox) have to extend the list of supported breakpoints – `extra-small` and `extra-large` have to be added (for now it is only `small`, `medium`, and `large`)
```
Comment (sashuk) [inline docs/decisions/13-breakpoints.md:55]: More detailed step

File: docs/decisions/14-page-layout.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+BASE design standardises page layout – the page content (section between header bar and bottom of the page) is expected to have only three possible layouts (please see the [Figma link](https://www.figma.com/file/q2nvjiyO2CLqBv4DeJnU3i/Product-Library-Documentation?node-id=4788%3A37925&t=Qekpwfkixg6Feqpq-1) for more details):
+
+- page content with full sidebar – for screens equal or wider than `1280px`
+- page content with compact sidebar – for screens equal or wider than `1280px`
+- page content without sidebar (or with sidebar, accessible via hamburger button) – for screens smaller than `1280px`
+
+## Decision
+
+- Sidebar is hidden (and is accessible via hamburger button) on screens narrower than `1280px`
+- Sidebar is shown on screens equal or wider than `1280px` and it has two variants – wide and compact
```
Comment (dmaklygin) [inline docs/decisions/14-page-layout.md:18]: this is not clear, we can use both variants for each screen (`1280px` and `1280px+`) OR the variants belongs to screen sizes respectively?

Comment (sashuk) [reply docs/decisions/14-page-layout.md:18]: @dmaklygin Clarified this, thanks!

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Review by dmaklygin — APPROVED** [review]
👍

**Comment by sashuk** [comment]
Merging due to 👍's in https://toptal-core.slack.com/archives/C052MRF4QJV/p1681838691318619

**Pull request merged**